styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -171,7 +193,9 @@ private void ensureSecret(WorkflowInstance workflowInstance, RunSpec runSpec) {
                                     .build());
         LOG.info("[AUDIT] Secret {} created for service account {}", secretName, serviceAccount);
       }
-    } else if (runSpec.secret().isPresent()) {
+    }
+
+    if (runSpec.secret().isPresent()) {
```

Comment:
That can still be achieved by having some of the workflows using `service_account` while others using `secret`, in the same component. It feels weird having both. If the user doesn't feel like to migrate now because they have other keys stored in kube, they don't have to migrate and everything should still work.